### ISSUE TITLE
Message-consumer ignores the metadata given to `withMetadata` 

### ISSUE THREAD

**Issue by dimkin-eu** [pact-foundation/pact-js#693]
### Checklist

Before making a feature request, I have: 

- [x] [Searched the issues to check that this feature hasn't been requested before](https://github.com/pact-foundation/pact-js/issues?q=is%3Aissue)
- [x] Checked the documentation to see if it is possible to do what I want already 

### Feature description

in message consumer example only body is verified now, could be nice to do the same also with metadata
To be more specific - [dog handler](https://github.com/pact-foundation/pact-js/blob/master/examples/messages/consumer/dog-handler.ts) must verify also metadata [from spec](https://github.com/pact-foundation/pact-js/blob/master/examples/messages/consumer/message-consumer.spec.ts)

### Use case

as this info is in the pact and could be verified by providers, could be nice, that it is done also on the js consumer side.

**Comment by TimothyJones** [comment]
I’m sorry, I don’t understand what you’re asking for. Could you elaborate a bit?

**Comment by dimkin-eu** [comment]
@TimothyJones , no worries. `dogApiHandler` verifies only `withContent` part, but `withMetadata` is not (it's not even passed to handler)

**Comment by TimothyJones** [comment]
@mefellows : Do you know what the metadata field is for? It's not used anywhere in the source as far as I can see (except for where it's set).

**Comment by mefellows** [comment]
The metadata is not actually used on the consumer side, or at least, it's not currently exposed to it. It _may_ be used on the provider side, to verify things like message headers. 

Classic example would be the queue or topic it's published on.

Pact JS does not currently have the ability to validate message metadata, only Pact JVM does at the present time.

**Comment by TimothyJones** [comment]
Ah, right. It seems that Pact JS currently allows you to specify the metadata in the consumer, but it's not written to the pact (as far as I can see, it's not even passed beyond the DSL layer).

**Comment by TimothyJones** [comment]
@dimkin-eu What do you think of something like this on the provider side:

```
  messageProviders: {
    'a request to save an event': providerWithMetadata(
      () => yourUsualProviderHere(),
      { // Arbitrary metadata here
        'content-type': 'application/json',
        messageQueueType: 'AWS-SQS',
      }
    ),
  },
```
?

**Comment by TimothyJones** [comment]
This is fixed in 10.0.0-beta.45, and will be released in the next major release.
